src/lib/Auth.svelte

```diff
@@ -0,0 +1,64 @@
+<script lang="ts" context="module">
+    import {invoke} from "@tauri-apps/api";
+    import {appWindow, WebviewWindow} from "@tauri-apps/api/window";
+    import {settings} from "./util/settings";
+
+    export async function login(): Promise<string | void> {
+        const envVars = await getEnvVariables()
+        if (!envVars) throw new Error("Env variables not set")
+
+        return invoke('log_in').then((port) => {
+            const webview = new WebviewWindow('Login', {
+                url: envVars.oidcBaseUrl + "/auth?scope=openid&response_type=code&client_id=" + envVars.oidcClientId + "&redirect_uri=http://localhost:" + port,
+                title: "NBAuth innlogging"
+            })
+            appWindow.listen('token_exchanged', (event) => {
+                setRefreshAccessTokenInterval()
+                settings.authResponse = event.payload as AuthenticationResponse
+                webview.close()
+            });
+        });
+    }
+
+    export async function refreshAccessToken(): Promise<AuthenticationResponse | void> {
+        const authResponse = await settings.authResponse
+
+        if (await canRefresh() && authResponse) {
+            return invoke<AuthenticationResponse>("refresh_token", {refreshToken: authResponse.tokenResponse.refreshToken})
+                .then((res) => {
+                    settings.authResponse = res
+                    setRefreshAccessTokenInterval()
```

Comment:
Setter vi nytt interval for hver gang vi refresher så bygger vi opp at det skal kjøre mer og mer requests. Så den vil vi nok bare kalle en gang. Enten ved login eller ved første automatiske refresh på oppstart.